spec/index.md

```diff
@@ -1954,6 +1954,15 @@ The optional `settings` parameter can be used to set the following settings:
 
     Default value: 5_000_000_000_000 (5 trillion cycles).
 
+-   `wasm_memory_limit` (`nat`)
+
+    Must be a number between 0 and 2<sup>32</sup>-1, inclusively, and indicates the upper limit on the WASM heap memory consumption of the canister.
```

Comment:
@mraszyk: would you mind adjusting this limit to 2^48 (256TB) similar to the exiting `memory_allocation` field. The reason: there is an upcoming wasm64 feature which will allow the wasm memory to be larger than 4GiB.

Reply:
How about the default value (currently 3GiB)? And how about canisters that do not declare a 64-bit memory - there using limits and defaults for the 64-bit case doesn't make much sense, right?

Reply:
For 64-bit canisters we will still have the limit on the replica side (initially it will be 6 or 8 GiB). We can set the default value based on that. Instead of hardcoding the default in the spec, would it make sense to use a named parameter?

Reply:
Discussed offline.